lib/ansible/modules/cloud/amazon/aws_acm.py

```diff
@@ -0,0 +1,403 @@
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This software is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this software.  If not, see <http://www.gnu.org/licenses/>.
+#
+# Author:
+#   - Matthew Davis <Matthew.Davis.2@team.telstra.com>
+#     on behalf of Telstra Corporation Limited
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = '''
+module: aws_acm
+short_description: Upload and delete certificates in the AWS Certificate Manager service
+description:
+  - Import and delete certificates in Amazon Web Service's Certificate Manager (AWS ACM).
+  - >
+    This module does not currently interact with AWS-provided certificates.
+    It currently only manages certificates provided to AWS by the user.
+  - The ACM API allows users to upload multiple certificates for the same domain name,
+    and even multiple identical certificates.
+    This module attempts to restrict such freedoms, to be idempotent, as per the Ansible philosophy.
+    It does this through applying AWS resource "Name" tags to ACM certificates.
+  - >
+    When C(state=present),
+    if there is one certificate in ACM
+    with a C(Name) tag equal to the C(name_tag) parameter,
+    and an identical body and chain,
+    this task will succeed without effect.
+  - >
+    When C(state=present),
+    if there is one certificate in ACM
+    a C(Name) tag equal to the C(name_tag) parameter,
+    and a different body,
+    this task will overwrite that certificate.
+  - >
+    When C(state=present),
+    if there are multiple certificates in ACM
+    with a C(Name) tag equal to the C(name_tag) parameter,
+    this task will fail.
+  - >
+    When C(state=absent) and C(certificate_arn) is defined,
+    this module will delete the ACM resource with that ARN if it exists in this region,
+    and succeed without effect if it doesn't exist.
+  - >
+    When C(state=absent) and C(domain_name) is defined,
+    this module will delete all ACM resources in this AWS region with a corresponding domain name.
+    If there are none, it will succeed without effect.
+  - >
+    When C(state=absent) and C(certificate_arn) is not defined,
+    and C(domain_name) is not defined,
+    this module will delete all ACM resources in this AWS region with a corresponding C(Name) tag.
+    If there are none, it will succeed without effect.
+  - Note that this may not work properly with keys of size 4096 bits, due to a limitation of the ACM API.
+version_added: "2.9"
```

Comment:
This needs to be updated to `2.10`, otherwise CI will complain on the next run.

Reply:
changed